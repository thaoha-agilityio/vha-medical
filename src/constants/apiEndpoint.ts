export const API_ENDPOINT = {
  AUTH: '/auth/local',
  APPOINTMENTS: '/appointments',
  NOTIFICATIONS: '/notifications',
  SPECIALTIES: '/specialties',
  USERS: '/users',
  UPLOAD: '/upload',
  CHEMISTS: '/chemists',
  PERMISSIONS: '/users-permissions',
};

export const API_URL = `${process.env.NEXT_PUBLIC_API_URL}/api`;
export const IMGBB_URL = `${process.env.NEXT_PUBLIC_IMGBB_URL}?key=${process.env.NEXT_PUBLIC_IMGBB_API_KEY}`;

export const AUTH_TOKEN = process.env.NEXT_PUBLIC_AUTH_TOKEN;
