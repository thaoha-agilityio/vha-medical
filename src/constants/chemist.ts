import { UserModel } from '@/types';

export const DEFAULT_CHEMIST_DATA: UserModel = {
  email: '',
  username: '',
  password: '',
  description: '',
  rating: 0,
  tasks: 0,
  reviews: 0,
  specialtyId: undefined,
  avatar: undefined,
};

export const CHEMISTS_SEARCH_PARAMS = ['specialtyId'] as const;
