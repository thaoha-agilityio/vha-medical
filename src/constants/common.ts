export const RESULT_NOT_FOUND = 'Result Not Found';
