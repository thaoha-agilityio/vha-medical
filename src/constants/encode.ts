export const ENCODE_COOKIE = {
  algorithm: 'aes-256-cbc',
  secret: process.env.COOKIE_SECRET,
};
