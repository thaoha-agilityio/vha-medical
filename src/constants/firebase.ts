export const REGISTRATION_TOKENS = 'token';
