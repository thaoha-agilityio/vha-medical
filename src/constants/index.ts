export * from './apiEndpoint';
export * from './appointment';
export * from './common';
export * from './firebase';
export * from './image';
export * from './message';
export * from './navigation';
export * from './pagination';
export * from './regex';
export * from './router';
export * from './timing';
export * from './chemist';
export * from './encode';
