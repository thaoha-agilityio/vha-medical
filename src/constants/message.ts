export const FORM_VALIDATION_MESSAGE = {
  REQUIRED: (item: string) => `${item} must not be empty.`,
  ALL_WHITE_SPACE: (item: string) => `${item} cannot contain only whitespace.`,
  INVALID: (item: string) => `${item} is invalid. Please try again.`,
  MIN_LENGTH: (item: string, min: number) =>
    `${item} must be at least ${min} characters.`,
  MAX_LENGTH: (item: string, max: number) =>
    `${item} must be less than ${max} characters.`,
  PASSWORD_NOT_MATCH: 'Password does not match',
  FORMAT: (ariaLabel: string) =>
    `${ariaLabel} does not follow the correct format.`,
  MIN_TIME: (item: string) => `${item} must be in present or future.`,
  DURATION: (item: string) => `${item} must be between 00:00 and 23:59.`,
  NOT_SAME_AS_SENDER: 'Sender and receiver cannot be the same.',
  ONLY_TEXT: 'Please enter text only.',
  MAX_SIZE: ({
    item,
    size,
    unit,
  }: {
    item: string;
    size: number;
    unit: string;
  }) => `Max ${item} size is ${size}${unit}`,
  ACCEPTED_FORMATS: 'Only .jpg, .jpeg, .png and .webp formats are supported.',
};

export const ERROR_MESSAGE = {
  LOGIN: 'Login failed. Please try again.',
  SIGNUP: 'Signup failed. Please try again.',
  DELETE: (item: string) => `Delete ${item} failed. Please try again.`,
  CREATE: (item: string, error: string) =>
    `Create ${item} failed. Detail error: ${error}.`,
  UPDATE: (item: string, error: string) =>
    `Update ${item} failed. Detail error: ${error}.`,
  CANCEL: (item: string) => `Cancel ${item} failed. Please try again.`,
  DUPLICATE_FIELD: 'This attribute must be unique',
  USERNAME: 'Username already taken',
};

export const SUCCESS_MESSAGE = {
  LOGIN: 'Login successful.',
  SIGNUP: 'Signup successful.',
  DELETE: (item: string) => `Delete ${item} successfully.`,
  CREATE: (item: string) => `Create ${item} successfully.`,
  UPDATE: (item: string) => `Update ${item} successfully.`,
  CANCEL: (item: string) => `Cancel ${item} successfully.`,
};

export const NOTIFICATION_CONTENT = (action: string) =>
  `have been ${action} appointment`;

export const EXCEPTION_ERROR_MESSAGE = {
  GET: (item: string) => `An unexpected error occurred when getting ${item}.`,
  ADD: (item: string) => `An unexpected error occurred when adding ${item}.`,
  UPDATE: (item: string) =>
    `An unexpected error occurred when updating ${item}.`,
  DELETE: (item: string) =>
    `An unexpected error occurred when deleting ${item}.`,
  CANCEL: (item: string) =>
    `An unexpected error occurred when canceling ${item}.`,
  LOGIN: 'An unexpected error occurred in the login request',
  REGISTER: 'An unexpected error occurred in the register request.',
};
