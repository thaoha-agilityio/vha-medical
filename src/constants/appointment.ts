import { AppointmentModel, AppointmentStatusOption } from '@/types';
import { DEFAULT_CHEMIST_DATA } from './chemist';

export const APPOINTMENT_STATUS_OPTIONS: AppointmentStatusOption[] = [
  {
    key: 'new',
    label: 'New',
    value: 0,
  },
  {
    key: 'meeting',
    label: 'Meeting',
    value: 1,
  },
  {
    key: 'cancelled',
    label: 'Cancelled',
    value: 2,
  },
];

export const APPOINTMENT_STATUS = [
  {
    key: '0',
    label: 'New',
  },
  {
    key: '1',
    label: 'Meeting',
  },
  {
    key: '2',
    label: 'Canceled',
  },
];

export const DEFAULT_APPOINTMENT_DATA: AppointmentModel = {
  startTime: '',
  durationTime: '',
  status: 0,
  receiverId: {
    data: {
      id: '',
      attributes: DEFAULT_CHEMIST_DATA,
    },
  },
  senderId: {
    data: {
      id: '',
      attributes: DEFAULT_CHEMIST_DATA,
    },
  },
};

export const APPOINTMENT_SEARCH_PARAMS = ['receiverId', 'senderId'] as const;
