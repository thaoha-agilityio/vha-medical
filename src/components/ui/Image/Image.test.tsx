import React from 'react';
import { render, fireEvent } from '@testing-library/react';

// Components
import { Image } from '.';

// Constants
import { SRC_IMAGE_NOT_AVAILABLE } from '@/constants';

// Mock useState
jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useState: jest.fn(),
}));

describe('Image Component', () => {
  const props = {
    width: 100,
    height: 100,
    className: 'mock-class',
    alt: 'Mock Alt',
    src: 'mock.src',
  };

  const ImageComponent = () => {
    return render(<Image {...props} alt="Mock Alt" />);
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('Match snapshot', () => {
    const useStateSpy = jest.spyOn(React, 'useState');

    useStateSpy.mockReturnValue([true, jest.fn()]);

    const { container } = render(
      <Image {...props} src={SRC_IMAGE_NOT_AVAILABLE} alt="Mock Alt" />,
    );

    expect(container).toMatchSnapshot();
  });

  it('Sets fallback source on error', () => {
    const setState = jest.fn();
    const useStateSpy = jest.spyOn(React, 'useState');

    useStateSpy.mockReturnValue([SRC_IMAGE_NOT_AVAILABLE, setState]);

    const { getByAltText } = ImageComponent();
    const image = getByAltText(props.alt);

    fireEvent.error(image);

    expect(setState).toHaveBeenCalledWith(SRC_IMAGE_NOT_AVAILABLE);
  });
});
