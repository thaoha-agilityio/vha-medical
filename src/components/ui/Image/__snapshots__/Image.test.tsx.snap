// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Component Match snapshot 1`] = `
<div>
  <img
    alt="Mock Alt"
    class="mock-class"
    data-nimg="1"
    decoding="async"
    fetchpriority="high"
    height="100"
    src="http://localhost/"
    style="object-fit: cover;"
    width="100"
  />
</div>
`;
